client/client.go

```diff
@@ -0,0 +1,63 @@
+package client
+
+import (
+	"context"
+	"fmt"
+	"net/http"
+	"net/url"
+
+	"github.com/fabric8-services/fabric8-auth-client/auth"
+	goaclient "github.com/goadesign/goa/client"
+	"github.com/goadesign/goa/middleware/security/jwt"
+	errs "github.com/pkg/errors"
+)
+
+func NewAuthClient(hostURL string) (*AuthClient, error) {
+	u, err := url.Parse(hostURL)
+	if err != nil {
+		return nil, err
+	}
+
+	client := http.Client{}
+	c := auth.New(&doer{
+		target: goaclient.HTTPClientDoer(&client),
+	})
+	c.Host = u.Host
+	c.Scheme = u.Scheme
+	return &AuthClient{c}, nil
+}
+
+type doer struct {
+	target goaclient.Doer
+}
+
+func (d *doer) Do(ctx context.Context, req *http.Request) (*http.Response, error) {
+	token := jwt.ContextJWT(ctx)
+	if token != nil {
+		req.Header.Set("Authorization", fmt.Sprintf("Bearer %s", token.Raw))
+	}
+	return d.target.Do(ctx, req)
+}
+
+type AuthClient struct {
+	*auth.Client
+}
+
+func (c *AuthClient) CheckSpaceScope(ctx context.Context, spaceID, requiredScope string) error {
+	resp, err := c.Client.ScopesResource(ctx, auth.ScopesResourcePath(spaceID))
```

Comment:
For now I think we can use ScopesResource() here but I would optimize it later (in common).
@sbryzak this is something we need to discuss with you when you are back :)